CMakeLists.txt

```diff
@@ -29,374 +44,351 @@ option(NO_MEMLOCK "Do not lock buffer memory (useful when running in Docker)" OF
 option(SUPERDEBUG "Enable extra debugging with no optimisation" OFF)
 option(SANITIZE "Enable clang sanitizers for testing" OFF)
 option(COMPILE_DOCS "Use Sphinx to compile documentation" OFF)
-option(BOOST_TESTS "Compile Boost C++ unit tests" OFF)
+option(WITH_TESTS "Compile testing library and boost C++ unit tests" OFF)
 option(_GOSSEC "Set buffer status messages to DEBUG log level" OFF)
 option(IWYU "Enable include-what-you-use and print suggestions to stderr" OFF)
+option(CCACHE "Use ccache to speed up the build" OFF)
+option(WERROR "Warnings are errors" OFF)
```

Comment:
Should we not have this on by default?

Reply:
The idea is that we turn on a lot of warnings, but only make them errors on the CI server. This way we can't merge stuff that generates warnings, we also see our local compiler spit out all of them, but we never get stopped in e.g. the deployment process, when suddenly a compiler decides to warn about something new.
I remember the last example being very annoying once in the past. Also this was the recommended way by one of the cmake people in a talk I saw somewhere...

Reply:
You make a good point, let's leave it `OFF` then.